dkms.in

```diff
@@ -2418,6 +2426,13 @@ parallel_jobs=${parallel_jobs:-$(get_num_cpus)}
 [[ "$parallel_jobs" = 0 ]] && parallel_jobs=""
 
 setup_kernels_arches "$action"
+
+# Source in configuration again. Support variables like kernelver in
+# sign_file/mok_signing_key/mok_certificate
+for i in /etc/dkms/framework.conf /etc/dkms/framework.conf.d/*.conf; do
+    [[ -e "${i}" ]] && . "$i" >/dev/null
+done
+
```

Comment:
Are we meant to add this hunk, or move the pre-existing one further up the call chain?

Reply:
In particular bear in mind that "kernelver" may be an array, so things will explode when used directly as seen in the original report aka "/usr/lib/modules/$kernelver/build/certs-local/signing_crt.crt".

Reply:
Looking at the original Arch packages - none of them have `/usr/lib/modules/$kernelver/build/certs-local`. And considering the variable can be an array, I'm inclined that we should:
 - merge the tests - they're looking good
 - explicitly document that variable expansion is not supported in the framework.conf file(s)
 - error out when the config is invalid (the original report IMHO)

Reply:
>     * explicitly document that variable expansion is not supported in the framework.conf file(s)

I see your concern. But as long as we are still using ```. /etc/dkms/framework.conf ```, it's impossible to prohibit users from abusing framework.conf. They may do [something much worse than variable expansion](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1019425#52).

Reply:
Indeed, that's one of my concerns. People treating the config file for their scripting needs and thus reaching into places they should not.

Have you considered using `safe_source` that we have in-tree - that should help with such blatant abuses right? I realise it's a bit of a hack (due to other reasons), but that's orthogonal to my point. Namely: it will reduce (block even) hacks like the linked. Hence people who thought about being upstream-friendly but was never a priority, will finally reach out.

Reply:
Oh, I didn't even know the existence of ```safe_source```. I will close this pull request and split it into smaller ones.